native/cocos/renderer/gfx-gles2/GLES2Commands.cpp

```diff
@@ -2981,6 +2981,41 @@ void cmdFuncGLES2BlitTexture(GLES2Device *device, GLES2GPUTexture *gpuTextureSrc
     device->blitManager()->draw(gpuTextureSrc, gpuTextureDst, regions, count, filter);
 }
 
+void cmdFuncGLES2CopyTextureToTexture(GLES2Device *device, GLES2GPUTexture *srcGPU, GLES2GPUTexture *dstGPU, uint32_t dstX, uint32_t dstY, const Rect *srcRect) {
+    CC_ASSERT_NONZERO(dstGPU->glTexture);
+    auto x = 0U, y = 0U, w = srcGPU->width, h = srcGPU->height;
+    if (srcRect) {
+        CC_ASSERT_LE(srcRect->x + srcRect->width, srcGPU->width);
+        CC_ASSERT_LE(srcRect->y + srcRect->height, srcGPU->height);
+        x = srcRect->x;
+        y = srcRect->y;
+        w = srcRect->width;
+        h = srcRect->height;
+    }
+    CC_ASSERT_LE(dstX + w, dstGPU->width);
+    CC_ASSERT_LE(dstY + h, dstGPU->height);
+
+    bool isTexture = srcGPU->glTexture;
+    GLuint glResource = isTexture ? srcGPU->glTexture : srcGPU->glRenderbuffer;
+
+    GLuint framebuffer = 0U;
+    auto cache = device->stateCache();
+    GL_CHECK(glGenFramebuffers(1, &framebuffer));
+    if (cache->glFramebuffer != framebuffer) {
+        GL_CHECK(glBindFramebuffer(GL_FRAMEBUFFER, framebuffer));
+        cache->glFramebuffer = framebuffer;
+    }
+    GL_CHECK(glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, srcGPU->glTarget, glResource, 0));
+
+    GLenum status;
+    GL_CHECK(status = glCheckFramebufferStatus(GL_FRAMEBUFFER));
+    CC_ASSERT_EQ(status, GL_FRAMEBUFFER_COMPLETE);
+
+    GL_CHECK(glBindTexture(dstGPU->glTarget, dstGPU->glTexture));
+    cache->glTextures[cache->texUint] = dstGPU->glTexture;
+    GL_CHECK(glCopyTexSubImage2D(dstGPU->glTarget, 0, dstX, dstY, x, y, w, h));
```

Comment:
Memoryleak here. need to invoke glDeleteFramebuffers and reset cache->glReadFramebuffer to 0.